packages/assets-controllers/src/TokenRatesController.ts

```diff
@@ -358,36 +360,55 @@ export class TokenRatesController extends PollingControllerV1<
       return;
     }
 
-    const newContractExchangeRates = await this.#fetchAndMapExchangeRates({
-      tokenContractAddresses,
-      chainId,
-      nativeCurrency,
-    });
+    if (chainId in this.#inProcessExchangeRateUpdate) {
```

Comment:
This assumes that the calls differ only in chain ID, but this is incorrect. They can also differ in chain ID or token addresses being queried for.

I'll fix this after I figure out why the tests are failing (they're failing for an unrelated reason).

Reply:
Mostly resolved by https://github.com/MetaMask/core/pull/3635/commits/5c1e57f3070c666275fc6c91fa833b7255a6bee1, but still waiting on #3639 to be merged first so that we don't need to include token addresses in this key